storage/src/database/tezedge_database.rs

```diff
@@ -139,16 +140,19 @@ impl<S: KVStoreKeyValueSchema> TezedgeDatabaseWithIterator<S> for TezedgeDatabas
 
 pub struct TezedgeDatabase {
     backend: Arc<TezedgeDatabaseBackend>,
+    log: slog::Logger,
 }
 
 impl TezedgeDatabase {
-    pub fn new(backend_option: TezedgeDatabaseBackendOptions) -> Self {
+    pub fn new(backend_option: TezedgeDatabaseBackendOptions, log: &Logger) -> Self {
```

Comment:
@vlad9486 
this is not recommended, to get reference and then just clone it, 
change pls. `log: &Logger` -> `log: Logger`, so that caller can decide to clone or give ownership